optimum/intel/openvino/modeling_visual_language.py

```diff
@@ -2331,6 +2334,467 @@ def preprocess_inputs(
         return inputs
 
 
+class _OVQwen2_5_VLForCausalLM(OVModelForVisualCausalLM):
+    additional_parts = ["vision_embeddings_merger"]
+
+    def __init__(
+        self,
+        language_model: ov.Model,
+        text_embeddings: ov.Model,
+        vision_embeddings: ov.Model,
+        config: PretrainedConfig = None,
+        device: str = "CPU",
+        dynamic_shapes: bool = True,
+        ov_config: Optional[Dict[str, str]] = None,
+        model_save_dir: Optional[Union[str, Path, TemporaryDirectory]] = None,
+        quantization_config: Union[OVWeightQuantizationConfig, Dict] = None,
+        **kwargs,
+    ):
+        super().__init__(
+            language_model=language_model,
+            text_embeddings=text_embeddings,
+            vision_embeddings=vision_embeddings,
+            config=config,
+            device=device,
+            dynamic_shapes=dynamic_shapes,
+            ov_config=ov_config,
+            model_save_dir=model_save_dir,
+            quantization_config=quantization_config,
+            **kwargs,
+        )
+        self.rope_deltas = None  # cache rope_deltas here
+
+        class Qwen2_5_VisionRotaryEmbedding(torch.nn.Module):
+            def __init__(self, dim: int, theta: float = 10000.0) -> None:
+                super().__init__()
+                inv_freq = 1.0 / (theta ** (torch.arange(0, dim, 2, dtype=torch.float) / dim))
+                self.register_buffer("inv_freq", inv_freq, persistent=False)
+
+            def forward(self, seqlen: int) -> torch.Tensor:
+                seq = torch.arange(seqlen, device=self.inv_freq.device, dtype=self.inv_freq.dtype)
+                freqs = torch.outer(seq, self.inv_freq)
+                return freqs
+
+        head_dim = config.vision_config.hidden_size // config.vision_config.num_heads
+        self._rotary_pos_emb = Qwen2_5_VisionRotaryEmbedding(head_dim // 2)
+
+    def get_rope_index(
```

Comment:
could you add a link to https://github.com/huggingface/transformers/blob/482d17be60d1836a7e787c5c3d69d4c3ed171ebc/src/transformers/models/qwen2_vl/modeling_qwen2_vl.py#L1451

Reply:
added